### ISSUE TITLE
Feature: `coder config-ssh` to add "Include coder/config" to `~/.ssh/config`

### ISSUE THREAD

**Issue by michaelbrewer** [coder/coder#12390]
By default `coder config-ssh` writes to `~/.ssh/config`, the problem is that many other tools also writes there and ends up being quite messy.

Ideally it should edit a separate file like `~/.ssh/coder/config` and then have an "Include" to point to 

```config
## START CODER INTEGRATION
Include "coder/config"
## END CODER INTEGRATION
```

## Alternatives

Currently you can  use `--ssh-config-file`

```bash
coder config-ssh --ssh-config-file ~/.ssh/coder-config
```

But then you can use this config file or update `~/.ssh/config`

```bash
ssh -F ~/.ssh/coder/config coder.test
```
So having a convenience command line option to add the missing "Include" to `~/.ssh/config`

**Comment by kylecarbs** [comment]
Good idea! It's a nice idea to be able to `ssh -F`, and that way it could be excluded from the global scope if desired.

**Comment by michaelbrewer** [comment]
Right less things to debug when your local config just polluted by conflicting settings.

**Comment by stirby** [comment]
I'm slotting this in next script, related to Coder config storage in the CLI so linking https://github.com/coder/coder/issues/13325.

**Comment by mafredri** [comment]
We went down the path of using `Include` at one point which caused issues with certain tools. For the best compatibility, we should retain the current behavior IMO.

Personally I use something along the lines of this:

```
# ~/.ssh/config
Include coder-*

# ~/.zshrc
coder_deployment() {
    deployment=$1
    export CODER_CONFIG_DIR="$HOME/Library/Application Support/coderv2-${deployment}"
    export CODER_SSH_CONFIG_FILE="$HOME/.ssh/coder-${deployment}"
}
```

An important detail is also the placement of the `Include`, most likely it needs to be at the top of the file, and also moveable by the user. I'm not entirely opposed to adding an option to automatically add the `Include`, though, but it adds a bit of additional complexity we need to handle (granted, in a similar manner we already handle the config).
